### PR TITLE
Rl action

### PR DESCRIPTION
**Pull request by Smyther-Hsiao**
Added the first version of the action network used in RL.

### CODE REVIEW

File: applications/harnesses/lsc_action/README.md
Comment (hickmank) [inline applications/harnesses/lsc_action/README.md:1]: Update this README to explain what this harness does. It doesn't have to be long winded.

File: applications/harnesses/lsc_action/START_study.py
Comment (hickmank) [inline applications/harnesses/lsc_action/START_study.py:1]: This should be a symbolic link to `applications/harnesses/START_study.py`

File: applications/harnesses/lsc_action/hyperparameters.csv
Comment (hickmank) [inline applications/harnesses/lsc_action/hyperparameters.csv:1]: You don't actually use the variables `YOKE_CONDA_BASE`, `LIST_DIR`, `DESIGN_DIR`, `NPZ_DIR`. You should remove those columns from this.

File: applications/harnesses/lsc_action/train_lsc_action.py [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+    # Model Parameters
+    featureList = args.featureList
+    linearFeatures = args.linearFeatures
+
+    # Training Parameters
+    initial_learningrate = args.init_learnrate
+    LRepoch_per_step = args.LRepoch_per_step
+    LRdecay = args.LRdecay
+    batch_size = args.batch_size
+    terminal_steps = args.terminal_steps
```
Comment (hickmank) [inline applications/harnesses/lsc_action/train_lsc_action.py:102]: You don't use these steps that have to do with the LR-scheduler. Let's remove them.

File: applications/harnesses/lsc_action/train_lsc_action.py [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+)
+parser = argparse.ArgumentParser(
+    prog="LSC Surrogate Training", description=descr_str, fromfile_prefix_chars="@"
+)
+parser = cli.add_default_args(parser=parser)
+parser = cli.add_filepath_args(parser=parser)
+parser = cli.add_computing_args(parser=parser)
+parser = cli.add_model_args(parser=parser)
+parser = cli.add_training_args(parser=parser)
+# parser = cli.add_step_lr_scheduler_args(parser=parser)  # SH
```
Comment (hickmank) [inline applications/harnesses/lsc_action/train_lsc_action.py:68]: These two scheduler lines can be deleted.

File: applications/harnesses/lsc_action/train_lsc_action.py [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+        model = nn.DataParallel(model)
+
+    model.to(device)
+
+    for state in optimizer.state.values():
+        for k, v in state.items():
+            if isinstance(v, torch.Tensor):
+                state[k] = v.to(device)
+
+    #############################################
```
Comment (hickmank) [inline applications/harnesses/lsc_action/train_lsc_action.py:201]: You can delete all of these "scheduler related" lines.

File: applications/harnesses/lsc_action/train_lsc_action.py [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+            optimizer=optimizer,
+            loss_fn=loss_fn,
+            epochIDX=epochIDX,
+            train_per_val=train_per_val,
+            train_rcrd_filename=trn_rcrd_filename,
+            val_rcrd_filename=val_rcrd_filename,
+            device=device,
+        )
+
+        # Increment LR scheduler
```
Comment (hickmank) [inline applications/harnesses/lsc_action/train_lsc_action.py:310]: You can delete the scheduler lines.

File: applications/harnesses/lsc_action/training_START.input [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+<F0>
+<F1>
+<F2>
+<F3>
+<F4>
+--linearFeatures
+<LINEAR_F>
+--init_learnrate
+<LEARN_RATE>
+--LRepoch_per_step
```
Comment (hickmank) [inline applications/harnesses/lsc_action/training_START.input:27]: Lines related to LR-scheduler can be removed.

File: applications/harnesses/lsc_action/training_input.tmpl [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+<F0>
+<F1>
+<F2>
+<F3>
+<F4>
+--linearFeatures
+<LINEAR_F>
+--init_learnrate
+<LEARN_RATE>
+--LRepoch_per_step
```
Comment (hickmank) [inline applications/harnesses/lsc_action/training_input.tmpl:27]: Lines related to LR-scheduler can be removed.

File: applications/harnesses/lsc_action/training_slurm.tmpl [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+#SBATCH --output=study<studyIDX>_epoch<epochIDX>.out
+#SBATCH --error=study<studyIDX>_epoch<epochIDX>.err
+#SBATCH -vvv
+
+# Check available GPUs
+sinfo  -o "%P %.24G %N"
+srun /usr/bin/echo $CUDA_AVAILABLE_DEVICES
+
+# Load correct conda environment
+# source <YOKE_CONDA_BASE>/bin/activate <YOKE_TORCH_ENV>
```
Comment (hickmank) [inline applications/harnesses/lsc_action/training_slurm.tmpl:30]: Don't need this line anymore.

**Review by hickmank — CHANGES_REQUESTED** [review]
Minor changes requested.

**Comment by hickmank** [comment]
@Smyther-Hsiao let's get this PR merged in. Let me know if you need help.
